app/controllers/webhook_deliveries_controller.rb

```diff
@@ -4,13 +4,18 @@ def index
     size = params[:size] || 10
     client_application_id = params[:client_application_id]
 
-    response = ClientApplicationService.fetch_webhook_deliveries(
-      session[:access_token],
-      client_application_id: client_application_id, page: page, size: size
-    )
+    cache_key = "webhook_deliveries/#{client_application_id}/page/#{page}/size/#{size}"
+
+    # Use Rails.cache to cache the API response across requests
+    @webhook_deliveries = Rails.cache.fetch(cache_key, expires_in: 10.minutes) do
```

Comment:
Added a little cache layer so we dont make API calls when clicking back and forth between show and list pages